src/dft/descriptor_config_helper.hpp

```diff
@@ -0,0 +1,319 @@
+        }
+        std::copy(set_val, set_val + vals.rank + 1, vals.output_strides.begin());
+    }
+};
+
+template <precision prec, domain dom>
+struct set_value_helper<config_param::FWD_DISTANCE, prec, dom> {
+    static void set_val(
+        dft_values<prec, dom>& vals,
+        param_type_helper_t<real_helper_t<prec>, config_param::FWD_DISTANCE>& set_val) {
+        vals.fwd_dist = set_val;
+    }
+};
+
+template <precision prec, domain dom>
+struct set_value_helper<config_param::BWD_DISTANCE, prec, dom> {
+    static void set_val(
+        dft_values<prec, dom>& vals,
+        param_type_helper_t<real_helper_t<prec>, config_param::BWD_DISTANCE>& set_val) {
+        vals.bwd_dist = set_val;
+    }
+};
+
+template <precision prec, domain dom>
+struct set_value_helper<config_param::WORKSPACE, prec, dom> {
+    static void set_val(
+        dft_values<prec, dom>& vals,
+        param_type_helper_t<real_helper_t<prec>, config_param::WORKSPACE>& set_val) {
+        if (set_val == config_value::ALLOW || set_val == config_value::AVOID) {
+            vals.workspace = set_val;
+        }
+        else {
+            throw mkl::invalid_argument("DFT", "set_value", "Workspace must be allow or avoid.");
+        }
+    }
+};
+
+template <precision prec, domain dom>
+struct set_value_helper<config_param::ORDERING, prec, dom> {
+    static void set_val(dft_values<prec, dom>& vals,
+                        param_type_helper_t<real_helper_t<prec>, config_param::ORDERING>& set_val) {
+        if (set_val == config_value::ORDERED || set_val == config_value::BACKWARD_SCRAMBLED) {
+            vals.ordering = set_val;
+        }
+        else {
+            throw mkl::invalid_argument("DFT", "set_value",
+                                        "Ordering must be ordered or backwards scrambled.");
+        }
+    }
+};
+
+template <precision prec, domain dom>
+struct set_value_helper<config_param::TRANSPOSE, prec, dom> {
+    static void set_val(
+        dft_values<prec, dom>& vals,
+        param_type_helper_t<real_helper_t<prec>, config_param::TRANSPOSE>& set_val) {
+        vals.transpose = set_val;
+    }
+};
+
+template <precision prec, domain dom>
+struct set_value_helper<config_param::PACKED_FORMAT, prec, dom> {
+    static void set_val(
+        dft_values<prec, dom>& vals,
+        param_type_helper_t<real_helper_t<prec>, config_param::PACKED_FORMAT>& set_val) {
+        if (set_val == config_value::CCE_FORMAT) {
+            vals.packed_format = set_val;
+        }
+        else {
+            throw mkl::invalid_argument("DFT", "set_value", "Packed format must be CCE.");
+        }
+    }
+};
+
+template <config_param Param, precision prec, domain dom>
+void set_value(dft_values<prec, dom>& vals,
+               param_type_helper_t<real_helper_t<prec>, Param> set_val) {
+    // Note, set_val argument cannot be a ref for use with va_arg.
+    set_value_helper<Param, prec, dom>::set_val(vals, set_val);
```

Comment:
why not just a big 
```
static_assert(Param !=  config_param::PRECISION, "Read-Only parameter");
if constexpr (Param == config_param::FORWARD_SCALE) {
...
} else if constexpr (...
```

Reply:
Done in [24b7513](https://github.com/oneapi-src/oneMKL/pull/259/commits/24b7513ab591e41bcbdbcd0307956c3cc8289703).